catroid/src/main/java/org/catrobat/catroid/common/Survey.java

```diff
@@ -0,0 +1,197 @@
+import static org.catrobat.catroid.common.SharedPreferenceKeys.SURVEY_URL1_HASH_KEY;
+import static org.catrobat.catroid.common.SharedPreferenceKeys.SURVEY_URL2_HASH_KEY;
+import static org.catrobat.catroid.common.SharedPreferenceKeys.TIME_SPENT_IN_APP_KEY;
+
+public class Survey implements GetSurveyTask.SurveyResponseListener {
+	@VisibleForTesting
+	public static final int MINIMUM_TIME_SPENT_IN_APP = 3600;
+
+	long totalTimeSpentInApp;
+	long currentTimeSpentInApp;
+	long totalTimeSpentInStage;
+	long currentTimeSpentInStage;
+	boolean fulfilledSurveyRequirements;
+	boolean uploadFlag;
+
+	public Survey(Context context) {
+		fulfilledSurveyRequirements = false;
+		uploadFlag = false;
+		totalTimeSpentInStage = 0;
+		currentTimeSpentInApp = 0;
+
+		checkSurveyRequirement(context);
+	}
+
+	public void startAppTime(Context context) {
+		checkSurveyRequirement(context);
+
+		PreferenceManager.getDefaultSharedPreferences(context)
+				.edit()
+				.putLong(LAST_USED_DATE_KEY, new Date(System.currentTimeMillis()).getTime())
+				.apply();
+
+		if (!fulfilledSurveyRequirements && currentTimeSpentInApp == 0) {
+			currentTimeSpentInApp = System.currentTimeMillis();
+		}
+	}
+
+	public void endAppTime(Context context) {
+		if (!fulfilledSurveyRequirements) {
+			totalTimeSpentInApp = (System.currentTimeMillis() - currentTimeSpentInApp - totalTimeSpentInStage) / 1000;
+			currentTimeSpentInApp = 0;
+
+			long oldTimeSpentInApp = PreferenceManager.getDefaultSharedPreferences(context)
+					.getLong(TIME_SPENT_IN_APP_KEY, 0);
+
+			PreferenceManager.getDefaultSharedPreferences(context)
+					.edit()
+					.putLong(TIME_SPENT_IN_APP_KEY, oldTimeSpentInApp + totalTimeSpentInApp)
+					.apply();
+		}
+	}
+
+	public void startStageTime() {
+		if (!fulfilledSurveyRequirements) {
+			currentTimeSpentInStage = System.currentTimeMillis();
+		}
+	}
+
+	public void endStageTime() {
+		if (!fulfilledSurveyRequirements) {
+			totalTimeSpentInStage += System.currentTimeMillis() - currentTimeSpentInStage;
+		}
+	}
+
+	private void checkSurveyRequirement(Context context) {
+		if (!fulfilledSurveyRequirements) {
+			long timeSpentInApp = PreferenceManager.getDefaultSharedPreferences(context)
+					.getLong(TIME_SPENT_IN_APP_KEY, 0);
+
+			fulfilledSurveyRequirements = timeSpentInApp > MINIMUM_TIME_SPENT_IN_APP;
+		}
+	}
+
+	@VisibleForTesting
+	public boolean getUploadFlag() {
+		return uploadFlag;
+	}
+
+	public void setUploadFlag(boolean value) {
+		uploadFlag = value;
+	}
+
+	public void showSurvey(Context context) {
+		if (fulfilledSurveyRequirements) {
+			long oldDate = PreferenceManager.getDefaultSharedPreferences(context)
+					.getLong(LAST_USED_DATE_KEY, new Date(System.currentTimeMillis()).getTime());
+
+			boolean showSurvey = PreferenceManager.getDefaultSharedPreferences(context)
+					.getBoolean(SHOW_SURVEY_KEY, false);
+
+			if (!DateUtils.isToday(oldDate) || (showSurvey && getUploadFlag())) {
```

Comment:
I am not sure about this condition. To show the survey, how come `DateUtils.isToday(oldDate)` is allowed to be true, since the conditions are joined by a `||`?